docs/chart-distributions-ui.js

```diff
@@ -0,0 +1,172 @@
+    setInputValueFromSearchParam(timeUnitSelect);
+    TRANSFORMATION = timeUnitToNanos(timeUnitSelect.value);
+    timeUnitSelect.addEventListener("change", event => {
+        let timeUnit = event.target.value;
+        maxValueInput.value = timeUnitToMaxValue[timeUnit];
+        TRANSFORMATION = timeUnitToNanos(timeUnit);
+    
+        let input = event.target;
+        setURLSearchParam(input.name, input.value);
+        buildChartFromInputs();
+    })
+}
+
+// Open custom data modal when custom data option is selected
+const customDataInput = document.getElementById("custom-data-input-group");
+customDataInput && customDataInput.addEventListener('click', () => {
+    customDataModalOverlay.style.display = "block";
+    const customDataTextarea = document.querySelector("#custom-data-modal textarea");
+    if (!customDataTextarea.value) fillUpTextareaWithDummyData(customDataTextarea);
+})
+
+// Rebuild chart everytime the custom data text is changed
+const customDataTextarea = document.querySelector("#custom-data-modal textarea");
+customDataTextarea && customDataTextarea.addEventListener("change", () => buildChartFromInputs());
+
+// Close modal when we click the overlay
+const customDataModalOverlay = document.getElementById("custom-data-modal-overlay");
+customDataModalOverlay && customDataModalOverlay.addEventListener('click', () => {
+    customDataModalOverlay.style.display = "none";
+});
+
+// We need to stop propagation for click events on the actual modal,
+// so that clicking it doesn't close it
+const customDataModal = document.getElementById("custom-data-modal");
+customDataModal && customDataModal.addEventListener("click", event => event.stopPropagation());
+
+const options = document.querySelectorAll("#data-options input");
+options.forEach(option => {
+    option.addEventListener("change", event => {
+        event.preventDefault();
+
+        let input = event.target;
+        setURLSearchParam(input.name, input.value);
+        buildChartFromInputs();
+    });
+
+    if (searchParams().get(option.name) == option.value) {
+        option.checked = true;
+
+        // We won't save the custom data in the URL,
+        // if that is the value on load, we create dummy data
+        if (option.value == "custom") {
+            const customDataTextarea = document.querySelector("#custom-data-modal textarea");
+            fillUpTextareaWithDummyData(customDataTextarea);
+        }
+    }
+});
+
+const histogramKindSelector = document.querySelector("#histogram-props select#kind")
+const inputs = histogramKindSelector ? [
+    ...document.querySelectorAll("#histogram-props input"),
+    document.querySelector("#histogram-props select#kind")
+] : [];
+
+inputs.forEach(input => {
+    setInputValueFromSearchParam(input);
+    input.addEventListener("change", event => {
+        let input = event.target;
+        setURLSearchParam(input.name, input.value);
+        buildChartFromInputs();
+    });
+});
+
+buildChartFromInputs();
```

Comment:
This file is now included by default on all pages.
We should detect if there is any graph and otherwise bail out early to avoid any work if we know we don't display it.
We could wrap this in a function we can return from and put in a check on top

Reply:
Fixed this in eb70508, really should have done this from the beginning.